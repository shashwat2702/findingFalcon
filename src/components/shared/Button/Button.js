import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

const Button = ({ onClick, label, disabled }) => (
  <button
    className="editTaskButton"
    type="submit"
    onClick={onClick}
    disabled={disabled}
  >
    {label}
  </button>
);
Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
};
export default Button;
