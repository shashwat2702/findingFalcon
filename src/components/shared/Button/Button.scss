.editTaskButton {
    background-color: #0fd8a4;
    border: none;
    border-radius: 3px;
    height: 4rem;
    width: 12rem;
    font-size: 2rem;
    color: white;
}
.editTaskButton:disabled {
    background-color: #1a896c;
    border: none;
    border-radius: 3px;
    height: 4rem;
    width: 12rem;
    font-size: 2rem;
    color: #A9A9A9;
}