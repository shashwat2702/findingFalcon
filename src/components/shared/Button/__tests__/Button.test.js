import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';


jest.mock('react-router-dom');

describe('Button snapshot', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<Button onClick={jest.fn()} label="Search Falcon" />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
