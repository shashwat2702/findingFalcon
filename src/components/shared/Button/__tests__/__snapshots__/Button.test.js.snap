// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot should render correctly 1`] = `
<button
  className="editTaskButton"
  disabled={false}
  onClick={[MockFunction]}
  type="submit"
>
  Search Falcon
</button>
`;
